pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -759,21 +732,35 @@ func classifyClusterDecisions(decisions []fleetv1beta1.ClusterDecision) (selecte
 	return selected, unselected
 }
 
-func buildResourcePlacementStatus(crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot) []fleetv1beta1.ResourcePlacementStatus {
-	res := make([]fleetv1beta1.ResourcePlacementStatus, 0, len(latestSchedulingPolicySnapshot.Status.ClusterDecisions))
+func (r *Reconciler) setResourcePlacementStatusAndResourceConditions(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot, latestResourceSnapshot *fleetv1beta1.ClusterResourceSnapshot) error {
+	placementStatuses := make([]fleetv1beta1.ResourcePlacementStatus, 0, len(latestSchedulingPolicySnapshot.Status.ClusterDecisions))
 	decisions := latestSchedulingPolicySnapshot.Status.ClusterDecisions
 	selected, unselected := classifyClusterDecisions(decisions)
 
 	// In the pickN case, if the placement cannot be satisfied. For example, pickN deployment requires 5 clusters and
 	// scheduler schedules the resources on 3 clusters. We'll populate why the other two cannot be scheduled.
 	// Here it is calculating how many unscheduled resources there are.
 	unscheduledClusterCount := 0
-	if crp.Spec.Policy != nil &&
-		crp.Spec.Policy.PlacementType == fleetv1beta1.PickNPlacementType &&
-		crp.Spec.Policy.NumberOfClusters != nil {
-		unscheduledClusterCount = int(*crp.Spec.Policy.NumberOfClusters) - len(selected)
+	if crp.Spec.Policy != nil {
```

Comment:
I think the policy can be nil. what is the `unscheduledClusterCount` in that case?

Reply:
what's the semantic of nil policy? I thought it meant "applying to all clusters". What if the selected doesn't include all?